xla/pjrt/pjrt_stream_executor_client.cc

```diff
@@ -3393,7 +3405,32 @@ PjRtStreamExecutorClient::Compile(mlir::ModuleOp module,
       module, xla_computation,
       /*use_tuple_args=*/options.parameter_is_tupled_arguments,
       /*return_tuple=*/false));
-  return Compile(xla_computation, options);
+
+  TF_ASSIGN_OR_RETURN(std::vector<LayoutMode> arg_layout_modes,
+                      GetArgLayoutModes(module));
+  TF_ASSIGN_OR_RETURN(std::vector<LayoutMode> out_layout_modes,
+                      GetOutputLayoutModes(module));
+  TF_ASSIGN_OR_RETURN(std::vector<MemorySpaceColor> arg_memory_spaces,
+                      GetArgMemoryKinds(module));
+  TF_ASSIGN_OR_RETURN(std::vector<MemorySpaceColor> out_memory_spaces,
+                      GetOutputMemoryKinds(module));
+
+  Shape result_layout;
+  ExecutableBuildOptions build_options = options.executable_build_options;
+  TF_ASSIGN_OR_RETURN(auto arg_shapes_and_layouts,
```

Comment:
why is it called `arg_shapes_and_layouts`? It includes the results too right? Call it something like: `arg_and_result_layouts`?

Reply:
Results are buried somewhere in [build options](https://github.com/openxla/xla/blob/f052a450327bc4fb73dfd3685b6e4d176859d157/xla/pjrt/utils.cc#L705). Updated the name to `arg_layouts_and_pointers`.